extensions-core/azure-extensions/src/test/java/org/apache/druid/storage/azure/AzureStorageDruidModuleTest.java

```diff
@@ -163,120 +164,116 @@ public void testGetAzureEntityFactoryCanCreateAzureEntity()
     Object object2 = factory.create(cloudObjectLocation2, azureStorage, AzureInputSource.SCHEME);
     Object object3 = factory.create(cloudObjectLocation1, azureStorage, AzureStorageAccountInputSource.SCHEME);
-    Assert.assertNotNull(object1);
-    Assert.assertNotNull(object2);
-    Assert.assertNotNull(object3);
-    Assert.assertNotSame(object1, object2);
-    Assert.assertNotSame(object1, object3);
+    assertNotNull(object1);
+    assertNotNull(object2);
+    assertNotNull(object3);
+    assertNotSame(object1, object2);
+    assertNotSame(object1, object3);
   }
 
   @Test
   public void testGetAzureCloudBlobIteratorFactoryCanCreateAzureCloudBlobIterator()
   {
-    injector = makeInjectorWithProperties(PROPERTIES);
+    final Injector injector = makeInjectorWithProperties(PROPERTIES);
     AzureCloudBlobIteratorFactory factory = injector.getInstance(AzureCloudBlobIteratorFactory.class);
     Object object1 = factory.create(EMPTY_PREFIXES_ITERABLE, 10, azureStorage);
     Object object2 = factory.create(EMPTY_PREFIXES_ITERABLE, 10, azureStorage);
-    Assert.assertNotNull(object1);
-    Assert.assertNotNull(object2);
-    Assert.assertNotSame(object1, object2);
+    assertNotNull(object1);
+    assertNotNull(object2);
+    assertNotSame(object1, object2);
   }
 
   @Test
   public void testGetAzureCloudBlobIterableFactoryCanCreateAzureCloudBlobIterable()
   {
-    injector = makeInjectorWithProperties(PROPERTIES);
+    final Injector injector = makeInjectorWithProperties(PROPERTIES);
     AzureCloudBlobIterableFactory factory = injector.getInstance(AzureCloudBlobIterableFactory.class);
     AzureCloudBlobIterable object1 = factory.create(EMPTY_PREFIXES_ITERABLE, 10, azureStorage);
     AzureCloudBlobIterable object2 = factory.create(EMPTY_PREFIXES_ITERABLE, 10, azureStorage);
-    Assert.assertNotNull(object1);
-    Assert.assertNotNull(object2);
-    Assert.assertNotSame(object1, object2);
+    assertNotNull(object1);
+    assertNotNull(object2);
+    assertNotSame(object1, object2);
   }
 
   @Test
   public void testSegmentKillerBoundSingleton()
   {
     Injector injector = makeInjectorWithProperties(PROPERTIES);
     OmniDataSegmentKiller killer = injector.getInstance(OmniDataSegmentKiller.class);
-    Assert.assertTrue(killer.getKillers().containsKey(AzureStorageDruidModule.SCHEME));
-    Assert.assertSame(
+    assertTrue(killer.getKillers().containsKey(AzureStorageDruidModule.SCHEME));
+    assertSame(
         AzureDataSegmentKiller.class,
         killer.getKillers().get(AzureStorageDruidModule.SCHEME).get().getClass()
     );
-    Assert.assertSame(
+    assertSame(
         killer.getKillers().get(AzureStorageDruidModule.SCHEME).get(),
         killer.getKillers().get(AzureStorageDruidModule.SCHEME).get()
     );
   }
 
-  @Test
-  public void testMultipleCredentialsSet()
+  @ParameterizedTest
+  @MethodSource("propertiesWithMultipleCredentials")
```

Comment:
Rewrote this as a parametrized test. The `expectedException` usage must also have had a bug, because this change highlighted a mistake in the test; `useAzureCredentialsChain` is what is checked in code for the config, not `managedIdentityClientId`.